test/rules/avoid_returning_widgets.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N avoid_returning_widgets`
+
+import 'package:flutter/widgets.dart';
+
+class MyWidget extends StatelessWidget {
```

Comment:
Consider adding tests for methods inside a mixin and an extension:

```
mixin M {
  Widget myWidget() => Container(); // LINT
}
extension on Object {
  Widget myWidget() => Container(); // LINT
}
```

I believe that the intent is for those to also be linted, and that the current implementation is correct, but a test would be good. I don't think it's necessary to include more than one example; what matters is the kind of declaration they're in, not whether it's a method vs. getter, etc.

Reply:
Good point! I just added them!